src/Workspaces/CoreTest/SolutionTests/SolutionTests.cs

```diff
@@ -4508,5 +4510,206 @@ public void GetRelatedDocumentsDoesNotReturnOtherTypesOfDocuments()
+            Assert.Single(frozenCompilation1.SyntaxTrees);
+            Assert.True(frozenCompilation1.ContainsSyntaxTree(await frozenProject1.Documents.Single().GetSyntaxTreeAsync()));
+
+            var frozenProject2 = frozenSolution.GetProject(project2.Id);
+            Assert.Empty(frozenProject2.Documents);
+
+            var frozenCompilation2 = await frozenProject2.GetCompilationAsync();
+            Assert.Empty(frozenCompilation2.SyntaxTrees);
+        }
+
+        [Fact]
+        public async Task TestFrozenPartialSolution5()
+        {
+            using var workspace = WorkspaceTestUtilities.CreateWorkspaceWithPartialSemantics();
+            var project1 = workspace.CurrentSolution.AddProject("CSharpProject1", "CSharpProject1", LanguageNames.CSharp);
+            var project2 = project1.Solution.AddProject("CSharpProject2", "CSharpProject2", LanguageNames.CSharp);
+            project1 = project2.Solution.GetProject(project1.Id).AddDocument("Doc1", SourceText.From("class Doc1 { }")).Project;
+            project2 = project1.Solution.GetProject(project2.Id).AddDocument("Doc2", SourceText.From("class Doc2 { }")).Project;
+
+            project2 = project2.AddProjectReference(new(project1.Id));
+            project1 = project2.Solution.GetProject(project1.Id);
+
+            // Getting compilation from project2 should affect project 1 as there's a ptp relationship with it.
+            await project2.GetCompilationAsync();
+
+            var frozenSolution = await project1.Solution.WithFrozenPartialCompilationsAsync(CancellationToken.None);
+            var frozenProject1 = frozenSolution.GetProject(project1.Id);
+            Assert.Single(frozenProject1.Documents);
+
+            var frozenCompilation1 = await frozenProject1.GetCompilationAsync();
+            Assert.Single(frozenCompilation1.SyntaxTrees);
+            Assert.True(frozenCompilation1.ContainsSyntaxTree(await frozenProject1.Documents.Single().GetSyntaxTreeAsync()));
+
+            var frozenProject2 = frozenSolution.GetProject(project2.Id);
+            Assert.Single(frozenProject2.Documents);
+
+            var frozenCompilation2 = await frozenProject2.GetCompilationAsync();
+            Assert.Single(frozenCompilation2.SyntaxTrees);
+            Assert.True(frozenCompilation2.ContainsSyntaxTree(await frozenProject2.Documents.Single().GetSyntaxTreeAsync()));
+
+            Assert.Single(frozenCompilation2.References.Where(r => r is CompilationReference c && c.Compilation == frozenCompilation1));
+        }
+
+        [Fact]
+        public async Task TestFrozenPartialSolution6()
+        {
+            using var workspace = WorkspaceTestUtilities.CreateWorkspaceWithPartialSemantics();
+            var project1 = workspace.CurrentSolution.AddProject("CSharpProject1", "CSharpProject1", LanguageNames.CSharp);
+            project1 = project1.AddDocument("Doc1", SourceText.From("class Doc1 { }")).Project;
+
+            // If we freeze
```

Comment:
I think you might have forgotten to complete

Reply:
that is true :)